hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -494,10 +497,32 @@ private void handleRetry(Object cause) {
 
             if (invokeCount < MAX_FAST_INVOCATION_COUNT) {
                 // fast retry for the first few invocations
-                context.asyncExecutor.execute(this);
+                context.asyncExecutor.execute(new InvocationRetryTask());
```

Comment:
Does this invocation still need to implement Runnable? I like the cleanup in the sense that I don't care about a bit of litter in case of retrying and now you responsibility from the invocation class.